tests/Functional/Handler/EmailHandlerTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\SubmissionBundle\Tests\Functional\Handler;
+
+use EMS\FormBundle\FormConfig\FormConfig;
+use EMS\FormBundle\FormConfig\SubmissionConfig;
+use EMS\FormBundle\Submit\AbstractResponse;
+use EMS\SubmissionBundle\Handler\EmailHandler;
+use EMS\SubmissionBundle\Tests\Functional\AbstractFunctionalTest;
+use Swift_Events_SendEvent;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+final class EmailHandlerTest  extends AbstractFunctionalTest
+{
+    /** @var EmailHandler */
+    protected $emailHandler;
+    /** @var \Swift_Mailer */
+    protected $mailer;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->emailHandler = $this->container->get('functional_test.emss.emailhandler');
+        $this->mailer = $this->container->get('mailer');
+    }
+
+    public function testSubmitFormData(): void
+    {
+        $data = ['name' => 'David', 'email' => 'user1@test.test',];
+        $form = $this->formFactory->createBuilder(FormType::class, $data, [])
+            ->add('name', TextType::class)
+            ->add('email', EmailType::class)
+            ->getForm();
+
+        $endpoint = '{{ data.email }}';
+        $message = json_encode([
+            'from' => 'noreply@test.test',
+            'subject' => 'Test submission',
+            'body' => 'Hi my name is {{ data.name }}'
+        ]);
+
+        $handle = $this->handle($form, $endpoint, $message, function (Swift_Events_SendEvent $evt) {
+            $this->assertEquals(['user1@test.test'], array_keys($evt->getMessage()->getTo()));
+            $this->assertEquals(['noreply@test.test'], array_keys($evt->getMessage()->getFrom()));
+            $this->assertEquals('Test submission', $evt->getMessage()->getSubject());
+            $this->assertEquals('Hi my name is David', $evt->getMessage()->getBody());
+        });
+
+        $this->assertEquals('{"status":"success","data":"Submission send by mail."}', $handle->getResponse());
+    }
+
+    public function testSubmitMultipleFiles(): void
+    {
+        $data = ['name' => 'David', 'files' => [
+            new UploadedFile(__DIR__ . '/../../files/attachment.txt', 'attachment.txt', 'text/plain'),
+            new UploadedFile(__DIR__ . '/../../files/attachment2.txt', 'attachment2.txt', 'text/plain'),
+        ]];
+        $form = $this->formFactory->createBuilder(FormType::class, $data, [])
+            ->add('name', TextType::class)
+            ->add('files', FileType::class, ['multiple' => true])
```

Comment:
Should we not use our components in testing in stead of the Symfony components directly?

Reply:
No that's the responsibility of the form bundle